### PR TITLE
Fix Bug 4010 - API method to register does not return authentification token 

### LINKED ISSUE #4010: API method to register does not return authentification token
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour
It authentificates the user, returns sessionid, csrftoken and CVAT token

### Current Behaviour
It authentificates the user, returns sessionid, csrftoken

### PR DESCRIPTION
**Pull request by g-kartik**
### Motivation and context

fixes #4010

When email verification is turned off in the all-auth settings, the rest-register api does not return auth-token key. This pull request solves the issue by returning the key in this condition.
Changes:
- Modify get_response_data of RegisterViewEx in iam views to add a new response field 'key' 
- The value of key will depend on the email verification setting of all-auth. If email verification is 'none', its value will be request user's auth token key else it will be 'None'.

### How has this been tested?

Tests
- Created tests for 'rest_register' api located in cvat/iam/tests/test_rest_apis.py to verify its response, both in case of email verification being turned on and turned off.
- From the root directory ran `python manage.py test cvat.apps.iam.tests` to run the tests

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- ~~[ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- ~~[ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by bsekachev** [comment]
Hi, @g-kartik 

Thank your for the contribution!
Please expect delay in review because of New Year coming, and have a nice New Year celebration!

**Comment by g-kartik** [comment]
> Hi, @g-kartik 
> 
> Thank your for the contribution!
> Please expect delay in review because of New Year coming, and have a nice New Year celebration! 
> 

Okay, Thank you!

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## \[2.0.0] - Unreleased
 ### Added
-
+- Fixed to return auth token key when registration without email verification (<https://github.com/openvinotoolkit/cvat/pull/4092>)
```
Comment (azhavoro) [inline CHANGELOG.md:10]: Please move this line to `Fixed` section.

**Review by azhavoro — CHANGES_REQUESTED** [review]

**Review by azhavoro — APPROVED** [review]

**Review by nmanovic — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**